test/makeCompleteAction.test.js

```diff
@@ -161,4 +161,182 @@ describe("makeCompleteAction", () => {
       });
     });
   });
+
+  describe("with custom payload options", () => {
+    const customPayload = payload => payload;
+
+    describe("when given a customRequestPayload option", () => {
+      it("should throw an error if the customRequestPayload is not a function", () => {
+        const action = makeCompleteAction("products", {
+          customRequestPayload: "not a function"
+        });
+
+        expect(() => action("request", { test: 123 })).toThrow(
+          "options.customRequestPayload should be a function."
+        );
+      });
+
+      it("should return a request action with a custom payload", () => {
+        const action = makeCompleteAction("Products", {
+          customRequestPayload: customPayload
+        });
+
+        expect(action("request", { test: 123 })).toEqual({
+          type: "GET_PRODUCTS",
+          test: 123
+        });
+      });
+
+      it("should return a default payload for the success action", () => {
```

Comment:
These tests could be a bit overkill, but I like having them there to make sure that I am not accidentally breaking other actions with my new custom action

Reply:
I see!